src/Controllers/User/InvoiceController.php

```diff
@@ -68,11 +68,13 @@ public function detail(ServerRequest $request, Response $response, array $args):
         $invoice->update_time = Tools::toDateTime($invoice->update_time);
         $invoice->pay_time = Tools::toDateTime($invoice->pay_time);
         $invoice_content = json_decode($invoice->content);
+        $invoice_type = $invoice_contenttype ?? 'product';
```

Comment:
Typo.

Reply:
This is actually not used, and with your improvements this variable can be removed. The front end can hide "Pay with balance" option by the type.